src/shogun/kernel/GaussianKernel.h

```diff
@@ -167,13 +192,9 @@ class GaussianKernel: public ShiftInvariantKernel
 	 */
 	virtual float64_t distance(int32_t idx_a, int32_t idx_b) const;
 
-private:
-	/** register parameters and initialize with defaults */
-	void register_params();
-
 protected:
 	/** width */
-	float64_t m_log_width;
+	AutoValue<float64_t> m_log_width = AutoValueEmpty{};
```

Comment:
so all auto parameters now need to have this type?

Reply:
Yes, and need to be initialised as empty. I mean the AutoValueEmpty can be assigned in the parameter registration, which might be safer?

Reply:
but yea in a nutshell this type will handle serde correctly and put/get, as well as check when auto initialisation is required

Reply:
sounds good to me